unity/PitayaExample/Assets/Pitaya/PitayaBinding.cs

```diff
@@ -272,12 +272,13 @@ public static void Request(IntPtr client, string route, byte[] msg, uint reqtId,
 
             if (rc != PitayaConstants.PcRcOk)
             {
-                DLog(string.Format("request - failed to perform request {0}", RcToStr(rc)));
+                var rcStr = RcToStr(rc)
+                DLog(string.Format("request - failed to perform request {0}", rcStr));
 
                 WeakReference reference;
                 if (!Listeners.TryGetValue(client, out reference) || !reference.IsAlive) return;
                 var listener = reference.Target as IPitayaListener;
-                if (listener != null) listener.OnRequestError(reqtId, new PitayaError(PitayaConstants.PitayaInternalError, "Failed to send request"));
+                if (listener != null) listener.OnRequestError(reqtId, new PitayaError(rcStr, "Failed to send request"));
```

Comment:
What do you think about keep raising an InternalError within the code in the message?
```suggestion
                if (listener != null) listener.OnRequestError(reqtId, new PitayaError(PitayaConstants.PitayaInternalError, string.Format("Failed to send request: {0}", rcStr)));
```